java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -653,7 +653,7 @@ private static boolean isDiag() {
     //spawn the indexing into a separate thread..
     private boolean spawnIndexLoadedRepo(final RepositoryInfo repo) {
 
-        if (!RepositoryPreferences.isIndexDownloadEnabled() && repo.isRemoteDownloadable()) {
+        if (!RepositoryPreferences.isIndexDownloadEnabledEffective() && repo.isRemoteDownloadable()) {
```

Comment:
I don't think this is correct. This would disable local indexing too if downloads are off.

Reply:
I did not change the logic here. Local repositories will report `repo.isRemoteDownloadable()` with false and thus the whole block will not be entered, indexing will continue. It would be cleaner to user `! isLocal()` here, but as said, I did not touch that logic.

Reply:
you are right, i thought that was a different method.